ext/handlers/chatmember.go

```diff
@@ -0,0 +1,33 @@
+package handlers
+
+import (
+	"fmt"
+
+	"github.com/PaulSonOfLars/gotgbot/v2"
+	"github.com/PaulSonOfLars/gotgbot/v2/ext"
+	"github.com/PaulSonOfLars/gotgbot/v2/ext/handlers/filters"
+)
+
+type ChatMember struct {
+	Response Response
+	Filter   filters.ChatMember
+}
+
+func NewChatMember(f filters.ChatMember, r Response) ChatMember {
+	return ChatMember{
+		Response: r,
+		Filter:   f,
+	}
+}
+
+func (c ChatMember) CheckUpdate(b *gotgbot.Bot, u *gotgbot.Update) bool {
+	return u.ChatMember != nil && c.Filter != nil && c.Filter(u.ChatMember)
```

Comment:
Just a question; opinions are welcome.

If Filter is nil, then this handler is completely pointless, and will always return false. That could lead to unexpected, hard to debug behaviour.

So the question is; would we rather:
a) have this "error" be swallowed up and have no indication that there is a bad thing happening (ie, the `Filter != nil` check is present, like in this PR)
b) fail with a runtime error because we're trying to call a nil function (potentially bad if happens in prod; would hopefully be tested first)

Reply:
I would strongly advice against the code that can lead to potential runtime errors. One should never de-reference a nil pointer. we should make the filter callback mandatory.

Reply:
I definitely agree - the question is, how do we define a nil filter?

In the current case, a nil filter == always false, handler never gets triggered.

However, i would argue that if I'm adding a new handler, and I set a nil filter, I would expect that handler to catch EVERY update; since I am not filtering on it. Does that make sense?

Reply:
Thats why i said the filter must be there. Lets say we have a ChatMember.All filter which will allow all ChatMember updates. and it only makes sense for a ChatMember update handler to only handle chatMember updates.. or u can have it like this. In case the user's filter is nil. You should still filter out the ChatMember updates.

Reply:
I'm not sure i follow what you're saying, could you rephrase? 

In my mind, no filter == all chat_member updates. Because you haven't "filtered" out anything.

Reply:
Dude, that ChatMemberAll filter func exists for a reason, does its name isnt able to make sense? And when did i ever say "i dont want any updates" part?

Reply:
I'm trying to have a discussion with you. If you are unable to do that calmly and respectfully, please refrain from opening PRs in the future.

I dont see why the `ChatMemberAll` filter and a `nil` filter can't have the same functionality. As I have explained above, I dont think a `nil` filter should stop any handlers from executing, since none of the updates have been "filtered" out.

I wasn't quoting you. I was describing functionality of the current code, which I believe is the wrong choice, so was asking for **your opinions** on. Currently, if the filter is `nil`, that handler will never be triggered. AKA, "I dont want any updates". I don't see a situation where this might be a useful outcome, which is why I am arguing against it.

Reply:
> I dont see why the `ChatMemberAll` filter and a `nil` filter can't have the same functionality. As I have explained above, I dont think a `nil` filter should stop any handlers from executing, since none of the updates have been "filtered" out.

If you are already aware of it then why not just edit them in Suggested Changes?

Reply:
Because I was asking for your opinion, and trying to understand your thought process on having it return as false.

This was a discussion, not a request. As I clearly stated in the opening comment.